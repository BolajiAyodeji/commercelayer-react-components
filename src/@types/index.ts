import { CSSProperties } from 'react'

export interface GeneralComponent {
  className?: string
  style?: CSSProperties
}

export interface GeneralReducer<S, A> {
  (state: S, action: A): S
}
