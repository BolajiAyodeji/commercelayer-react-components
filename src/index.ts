export { default as CommerceLayer } from './components/CommerceLayer'
export { default as OrderContainer } from './components/OrderContainer'
export { default as PriceContainer } from './components/PriceContainer'
export { default as Price } from './components/Price'
export { default as VariantContainer } from './components/VariantContainer'
export { default as VariantSelector } from './components/VariantSelector'
export { default as QuantitySelector } from './components/QuantitySelector'
export { default as AddToCart } from './components/AddToCart'
export { default as AvailabilityContainer } from './components/AvailabilityContainer'
export { default as AvailabilityTemplate } from './components/AvailabilityTemplate'
export { default as GiftCardContainer } from './components/GiftCardContainer'
export { default as GiftCard } from './components/GiftCard'
export { default as GiftCardCurrencySelector } from './components/GiftCardCurrencySelector'
export { default as GiftCardInput } from './components/GiftCardInput'
export { default as MetadataInput } from './components/MetadataInput'
export { default as SubmitButton } from './components/SubmitButton'
export { default as ItemContainer } from './components/ItemContainer'
export { default as LineItemsCount } from './components/LineItemsCount'
export { default as LineItemsContainer } from './components/LineItemsContainer'
export { default as LineItem } from './components/LineItem'
export { default as LineItemImage } from './components/LineItemImage'
export { default as LineItemName } from './components/LineItemName'
export { default as LineItemPrice } from './components/LineItemPrice'
export { default as LineItemQuantity } from './components/LineItemQuantity'
export { default as LineItemRemove } from './components/LineItemRemove'
export { default as SubTotal } from './components/SubTotal'
export { default as Discount } from './components/Discount'
export { default as Shipping } from './components/Shipping'
export { default as Taxes } from './components/Taxes'
export { default as GiftCardPrice } from './components/GiftCardPrice'
export { default as Total } from './components/Total'
export { default as Checkout } from './components/Checkout'
export { default as Errors } from './components/Errors'
