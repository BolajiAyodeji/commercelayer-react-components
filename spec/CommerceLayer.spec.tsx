import React from 'react'
import { CommerceLayer } from '../src'
import renderer from 'react-test-renderer'
import PropTypes from 'prop-types'

test('<CommerceLayer/>', () => {
  expect.assertions(6)
  const component = renderer.create(
    <CommerceLayer accessToken="myAccessToken" endpoint="myEndpoint">
      Commerce Layer
    </CommerceLayer>
  )
  const tree = component.toJSON()
  const root = component.toTree()
  const rendered = root.rendered
  const proptypes = root.type['propTypes']
  expect(tree).toMatchSnapshot()
  expect(root.props.accessToken).toBe('myAccessToken')
  expect(root.props.endpoint).toBe('myEndpoint')
  expect(proptypes.children).toBe(PropTypes.node.isRequired)
  expect(proptypes.accessToken).toBe(PropTypes.string.isRequired)
  expect(proptypes.endpoint).toBe(PropTypes.string.isRequired)
})
