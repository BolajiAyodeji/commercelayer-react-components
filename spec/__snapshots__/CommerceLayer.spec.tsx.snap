// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commerce Layer Component 1`] = `"Commerce Layer"`;
